design-proposals/migration-policies.md

```diff
@@ -0,0 +1,227 @@
+Today migration settings can be configurable only on the cluster-wide scope by editing [KubevirtCR's spec](https://kubevirt.io/api-reference/master/definitions.html#_v1_kubevirtspec)
+or more specifically [MigrationConfiguration](https://kubevirt.io/api-reference/master/definitions.html#_v1_migrationconfiguration)
+CRD. 
+
+Several aspects (although not all) of migration behaviour that can be customized are:
+- Bandwidth
+- Auto-convergence
+- Post/Pre-copy
+- Max number of parallel migrations
+- Timeout
+
+This design proposal asks to generalize the concept of defining migration configurations so it would be
+possible to apply different configurations to specified groups of VMs. These are meant to be called "**Migration
+Policies**".
+
+## Motivation
+Today there is no way of applying different configurations for different groups of VMs. Such capability can be useful
+for a lot of different use cases on which there is a need to differentiate between different workloads. Differentiation
+of different configurations could be needed because different workloads are considered to be in different priorities,
+security segregation, workloads with different requirements, help to converge workloads which aren't migration-friendly,
+and many other reasons (see user story section below).
+
+## Goals
+The goal is to have a mechanism through which it will be possible to apply all relevant cluster-wide configurations
+to a custom specified group of VM. Also, at any given time every VM should deterministically know which migration policy
+it needs to obey to.
+
+Another goal is to allow the cluster administrator to define these policies without the participation of the VM's owner
+(as written in "Definition of Users" below).
+
+## Non Goals
+* Allowing users to transparently choose migration policies for their workloads
+* Preventing misconfigurations regarding overlapping selectors
+
+## Definition of Users
+This feature's most obvious user is the cluster administrator.
+
+VM owners/creators, by contrast, should not know or care about their migration configuration in most cases.
+Moreover, the design should take into account that one of the goals is to allow the cluster administrator to dynamically
+define and change existing policies with as little participation as possible from the VM owner.
+
+Another goal is to let the admin the possibility to expose different policies to the users in a way he can then
+use. For example, an admin can set two policies for fast/slow migration that will apply to any VM with a
+`migration-type: slow` or `migration-type: fast` and this way a user can apply different policies by defining
+the relevant label.
+
+## User Stories
+As an admin I would like to have different migration policies for different groups of VMs.
+* By "migration policies" I mean controlling one or more of the following migration aspects:
+    * Bandwidth
+    * Scheduling
+    * Storage
```

Comment:
I don't think that scheduling or storage is a good example, not sure how can we affect this. 
Instead, I'd suggest listing things like sensitivity to overall migration time, migration progress timeout, whether the VMI is safe to switch to post-copy migration. In the future, we can expand this list and control the number of migration threads,  dirty rate.

Reply:
Thanks! Will do